src/main/java/org/reaktivity/specification/socks/internal/SocksFunctions.java

```diff
@@ -89,9 +159,40 @@ public SocksBeginExBuilder typeId(
         }
 
         public SocksBeginExBuilder address(
-            String address)
+            String address) throws UnknownHostException
         {
-            this.beginExRW.address(address);
+            if (IPV4_ADDRESS_MATCHER.get().reset(address).matches())
+            {
+                final Matcher ipv4Matcher = IPV4_ADDRESS_MATCHER.get();
+                final byte[] ipv4AddressBytes = IPV4_ADDRESS_BYTES.get();
+                for (int i = 0; i < ipv4AddressBytes.length; i++)
+                {
+                    ipv4AddressBytes[i] = parseByte(ipv4Matcher.group(i + 1), 10);
+                }
+                beginExRW.address(b -> b.ipv4Address(s -> s.set(ipv4AddressBytes)));
+            }
+            else if (IPV6_STD_ADDRESS_MATCHER.get().reset(address).matches())
+            {
+                final byte[] addressBytes = IPV6_ADDRESS_BYTES.get();
+                final Matcher ipv6Matcher = IPV6_STD_ADDRESS_MATCHER.get();
+                Arrays.fill(addressBytes, (byte) 0);
+                for (int i = 0; i < ipv6Matcher.groupCount(); i++)
+                {
+                    String ipv6Group = ipv6Matcher.group(i + 1);
+                    fillInBytes(addressBytes, 2 * i, ipv6Group);
+                }
+                beginExRW.address(b -> b.ipv6Address(s -> s.set(addressBytes)));
+            }
+            else if (IPV6_HEX_COMPRESSED_VALIDATE_MATCHER.get().reset(address).matches())
```

Comment:
Yes, because for `IPV6_HEX_COMPRESSED_MATCHER`,  if some cases like `1:2:3:4:5:6:7::1:2:3:4:5:6:7` will return true. The `IPV6_HEX_COMPRESSED_VALIDATE_MATCHER` will validate the general case, then after validate,  we can separate different groups. Maybe I can optimize pattern `IPV6_HEX_COMPRESSED_MATCHER` pattern.

Reply:
Understood. In that case, suggest you change this to:
```
if (IPV6_HEX_COMPRESSED_VALIDATE_MATCHER.get().reset(address).matches() &&
   (IPV6_HEX_COMPRESSED_MATCHER.get().reset(address).matches()))
```

Reply:
I see, that could help eliminating the `IPV6_HEX_COMPRESSED_MATCHER` matching steps. Thanks.

Reply:
done